wandb/sdk/interface/interface.py

```diff
@@ -671,6 +664,10 @@ def _publish_tbdata(self, tbrecord: pb.TBRecord) -> None:
     def _publish_telemetry(self, telem: tpb.TelemetryRecord) -> None:
         raise NotImplementedError
 
+    @abstractmethod
+    def publish_environment(self, environment: pb.EnvironmentRecord) -> None:
+        raise NotImplementedError
```

Comment:
convention in this file is to have the private implementation raise the exception